x-pack/plugins/ml/server/models/job_service/job_caps/field_service.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { cloneDeep } from 'lodash';
+import { Request } from 'src/legacy/server/kbn_server';
+import { Field, Aggregation, FieldId } from '../../../../common/types/fields';
+import { ES_FIELD_TYPES } from '../../../../common/constants/field_types';
+import { rollupServiceProvider, RollupJob, RollupFields } from './rollup';
+import { aggregations } from './aggregations';
+import { CallWithRequestType } from '../../../client/elasticsearch_ml';
+
+const METRIC_AGG_TYPE: string = 'metrics';
+
+// TODO, is this list correct?
+const supportedTypes: string[] = [
+  ES_FIELD_TYPES.DATE,
+  ES_FIELD_TYPES.KEYWORD,
+  ES_FIELD_TYPES.TEXT,
+  ES_FIELD_TYPES.DOUBLE,
+  ES_FIELD_TYPES.INTEGER,
+  ES_FIELD_TYPES.FLOAT,
+  ES_FIELD_TYPES.LONG,
+  ES_FIELD_TYPES.BYTE,
+  ES_FIELD_TYPES.HALF_FLOAT,
+  ES_FIELD_TYPES.SCALED_FLOAT,
+  ES_FIELD_TYPES.SHORT,
+];
+
+export interface AggAndFieldList {
+  aggs: Aggregation[];
+  fields: Field[];
+}
+
+export function fieldServiceProvider(
+  indexPattern: string,
+  isRollup: boolean,
+  callWithRequest: CallWithRequestType,
+  request: Request
+) {
+  return new FieldsService(indexPattern, isRollup, callWithRequest, request);
+}
+
+class FieldsService {
```

Comment:
Should this file / class name match whatever name you go with for `x-pack/plugins/ml/public/services/field_service.ts`? `JobCapsService` ?

Reply:
this service is use to create the fields and aggregations and is only used internally in the newJobCaps function/endpoint. So i think the name FieldsService is applicable here.